src/shogun/kernel/Kernel.h

```diff
@@ -509,7 +509,7 @@ class Kernel : public SGObject
 		 */
 		virtual int32_t get_num_vec_lhs()
 		{
-			return num_lhs;
+			return lhs->get_num_vectors();
```

Comment:
what if lhs == nullptr?

Reply:
Hmm true, should it be something along the lines if(lhs) return lhs->get_num_vectors(); else return 0;

Reply:
well check what's the initial value of num_lhs and that should be returned if lhs is nullptr

Reply:
I think it's intialized by zero cause that's according to `/shogun/tests/unit/kernel/ProductKernel_unittest.cc` which is one of the failing SEGFAULT tests

`append_kernel` uses `adjust_num_lhs_rhs_initialized` function that basically checks if num_lhs isn't equal to zero if it is it sets it using `get_num_vec_lhs`

Reply:
init does the initialization... hence that should be returned (0) if lhs or rhs is not set

Reply:
@vigsterkr those segfaults errors are now fixed, how do you suggest I go about the other errors I have traced back the remaining errors, it's basically the same error over an over again say we take the `tests/unit/statistical_testing/internals/MultiKernelMMD_unittest.cc` test
on line 88 we starthttps://pastebin.com/dcntwX64 by constructing some new Gaussian vectors, which in turns calls `ShiftInvariantKernel::ShiftInvariantKernel(const std::shared_ptr<Features >&l, const std::shared_ptr<Features >&r) : Kernel(l, r, 0)` constructor which calls the kernel constructor setting `get_num_vec_lhs` and `get_num_vec_rhs` to be zero -since `lhs` and `rhs` are already nullptr.
Now the problem comes from the `inline float64_t kernel(int32_t idx_a, int32_t idx_b)` function inside kernel which has this require that would always break since idx_a is equal to zero and it has to be less than get_num_vec_lhs which in turn is equal to zero. anyways I will upload the pastebin so it may make a tiny bit more sense :grin: 
https://pastebin.com/dcntwX64

Reply:
sorry but it's really hard to debug by description. you'll have to try to resolve this yourself or create a very simple use-case where this specific bug is reproducible

Reply:
@vigsterkr Thanks I will work on this